File: proxy-gateway-client-go/client_test.go
package proxygatewayclient_test

import (
	"context"
	"encoding/json"
	"net/http"
	"net/http/httptest"
	"testing"
	"time"

	proxygatewayclient "github.com/traumwohnung/proxy-gateway/proxy-gateway-client-go"
)

func newTestServer(t *testing.T, sessions []proxygatewayclient.SessionInfo) *httptest.Server {
	t.Helper()
	mux := http.NewServeMux()

	mux.HandleFunc("GET /api/sessions", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(sessions)
	})

	mux.HandleFunc("GET /api/sessions/{username}", func(w http.ResponseWriter, r *http.Request) {
		want := sessions[0].Username
		if r.PathValue("username") != want {
			http.Error(w, `{"error":"no active session"}`, http.StatusNotFound)
			return
		}
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(sessions[0])
	})

	mux.HandleFunc("POST /api/sessions/{username}/rotate", func(w http.ResponseWriter, r *http.Request) {
		want := sessions[0].Username
		if r.PathValue("username") != want {
			http.Error(w, `{"error":"no active session"}`, http.StatusNotFound)
			return
		}
		updated := sessions[0]
		updated.LastRotationAt = time.Now().UTC()
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(updated)
	})

	return httptest.NewServer(mux)
}

// newUsageTestServer starts a test server whose /api/usage handler captures the
// last received request so tests can inspect query parameters, and returns a
// configurable response.
type usageServerOpts struct {
	response proxygatewayclient.UsageResponse
	status   int // 0 → 200
}

func newUsageTestServer(t *testing.T, opts usageServerOpts) (*httptest.Server, *http.Request) {
	t.Helper()
	var captured *http.Request
	if opts.status == 0 {
		opts.status = http.StatusOK
	}
	mux := http.NewServeMux()
	mux.HandleFunc("GET /api/usage", func(w http.ResponseWriter, r *http.Request) {
		captured = r
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(opts.status)
		if opts.status == http.StatusOK {
			json.NewEncoder(w).Encode(opts.response)
		} else {
			w.Write([]byte(`{"error":"bad request"}`))
		}
	})
	srv := httptest.NewServer(mux)
	t.Cleanup(srv.Close)
	return srv, captured
}

var testSession = proxygatewayclient.SessionInfo{
	SessionID:      1,
	Username:       "eyJzZXQiOiJyZXNpZGVudGlhbCIsIm1pbnV0ZXMiOjYwfQ==",
	ProxySet:       "residential",
	Upstream:       "198.51.100.1:6658",
	CreatedAt:      time.Now().UTC(),
	NextRotationAt: time.Now().Add(time.Hour).UTC(),
	LastRotationAt: time.Now().UTC(),
	Metadata:       map[string]any{"platform": "myapp"},
}

func TestListSessions(t *testing.T) {
	srv := newTestServer(t, []proxygatewayclient.SessionInfo{testSession})
	defer srv.Close()

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL, APIKey: "test"})
	sessions, err := c.ListSessions(context.Background())
	if err != nil {
		t.Fatalf("ListSessions: %v", err)
	}
	if len(sessions) != 1 {
		t.Fatalf("expected 1 session, got %d", len(sessions))
	}
	if sessions[0].ProxySet != "residential" {
		t.Errorf("expected proxy_set=residential, got %q", sessions[0].ProxySet)
	}
}

func TestGetSession(t *testing.T) {
	srv := newTestServer(t, []proxygatewayclient.SessionInfo{testSession})
	defer srv.Close()

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL, APIKey: "test"})

	got, err := c.GetSession(context.Background(), testSession.Username)
	if err != nil {
		t.Fatalf("GetSession: %v", err)
	}
	if got == nil {
		t.Fatal("expected session, got nil")
	}
	if got.Upstream != testSession.Upstream {
		t.Errorf("expected upstream %q, got %q", testSession.Upstream, got.Upstream)
	}

	// Non-existent username → nil, nil
	missing, err := c.GetSession(context.Background(), "notexist")
	if err != nil {
		t.Fatalf("GetSession (missing): %v", err)
	}
	if missing != nil {
		t.Error("expected nil for missing session")
	}
}

func TestForceRotate(t *testing.T) {
	srv := newTestServer(t, []proxygatewayclient.SessionInfo{testSession})
	defer srv.Close()

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL, APIKey: "test"})

	got, err := c.ForceRotate(context.Background(), testSession.Username)
	if err != nil {
		t.Fatalf("ForceRotate: %v", err)
	}
	if got == nil {
		t.Fatal("expected session after rotate, got nil")
	}

	// Non-existent username → nil, nil
	missing, err := c.ForceRotate(context.Background(), "notexist")
	if err != nil {
		t.Fatalf("ForceRotate (missing): %v", err)
	}
	if missing != nil {
		t.Error("expected nil for missing session")
	}
}

// ---------------------------------------------------------------------------
// QueryUsage tests
// ---------------------------------------------------------------------------

var testUsageResponse = proxygatewayclient.UsageResponse{
	Rows: []proxygatewayclient.UsageRow{
		{
			Proxyset:      strPtr("residential"),
			UploadBytes:   1024,
			DownloadBytes: 2048,
			TotalBytes:    3072,
		},
	},
	TotalCount: 1,
	Page:       1,
	PageSize:   100,
	TotalPages: 1,
}

func strPtr(s string) *string { return &s }

func TestQueryUsage_NoFilter(t *testing.T) {
	srv, _ := newUsageTestServer(t, usageServerOpts{response: testUsageResponse})

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL, APIKey: "test"})
	resp, err := c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{})
	if err != nil {
		t.Fatalf("QueryUsage: %v", err)
	}
	if resp.TotalCount != 1 {
		t.Errorf("want TotalCount=1, got %d", resp.TotalCount)
	}
	if len(resp.Rows) != 1 {
		t.Fatalf("want 1 row, got %d", len(resp.Rows))
	}
	if resp.Rows[0].UploadBytes != 1024 {
		t.Errorf("want upload=1024, got %d", resp.Rows[0].UploadBytes)
	}
}

func TestQueryUsage_SendsFromParam(t *testing.T) {
	var gotReq *http.Request
	mux := http.NewServeMux()
	mux.HandleFunc("GET /api/usage", func(w http.ResponseWriter, r *http.Request) {
		gotReq = r
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(proxygatewayclient.UsageResponse{Rows: []proxygatewayclient.UsageRow{}})
	})
	srv := httptest.NewServer(mux)
	defer srv.Close()

	from := time.Date(2026, 1, 15, 10, 0, 0, 0, time.UTC)
	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL})
	c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{From: from})

	if gotReq == nil {
		t.Fatal("no request received")
	}
	got := gotReq.URL.Query().Get("from")
	want := from.Format(time.RFC3339)
	if got != want {
		t.Errorf("from param: want %q, got %q", want, got)
	}
}

func TestQueryUsage_SendsToParam(t *testing.T) {
	var gotReq *http.Request
	mux := http.NewServeMux()
	mux.HandleFunc("GET /api/usage", func(w http.ResponseWriter, r *http.Request) {
		gotReq = r
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(proxygatewayclient.UsageResponse{Rows: []proxygatewayclient.UsageRow{}})
	})
	srv := httptest.NewServer(mux)
	defer srv.Close()

	to := time.Date(2026, 1, 16, 0, 0, 0, 0, time.UTC)
	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL})
	c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{To: to})

	got := gotReq.URL.Query().Get("to")
	want := to.Format(time.RFC3339)
	if got != want {
		t.Errorf("to param: want %q, got %q", want, got)
	}
}

func TestQueryUsage_SendsProxysetParam(t *testing.T) {
	var gotReq *http.Request
	mux := http.NewServeMux()
	mux.HandleFunc("GET /api/usage", func(w http.ResponseWriter, r *http.Request) {
		gotReq = r
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(proxygatewayclient.UsageResponse{Rows: []proxygatewayclient.UsageRow{}})
	})
	srv := httptest.NewServer(mux)
	defer srv.Close()

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL})
	c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{Proxyset: "residential"})

	if gotReq.URL.Query().Get("proxyset") != "residential" {
		t.Errorf("proxyset param not sent correctly, got %q", gotReq.URL.Query().Get("proxyset"))
	}
}

func TestQueryUsage_SendsMetaParam(t *testing.T) {
	var gotReq *http.Request
	mux := http.NewServeMux()
	mux.HandleFunc("GET /api/usage", func(w http.ResponseWriter, r *http.Request) {
		gotReq = r
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(proxygatewayclient.UsageResponse{Rows: []proxygatewayclient.UsageRow{}})
	})
	srv := httptest.NewServer(mux)
	defer srv.Close()

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL})
	c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{MetaContains: `{"user":"alice"}`})

	if gotReq.URL.Query().Get("meta") != `{"user":"alice"}` {
		t.Errorf("meta param not sent correctly, got %q", gotReq.URL.Query().Get("meta"))
	}
}

func TestQueryUsage_SendsGranularityParam(t *testing.T) {
	var gotReq *http.Request
	mux := http.NewServeMux()
	mux.HandleFunc("GET /api/usage", func(w http.ResponseWriter, r *http.Request) {
		gotReq = r
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(proxygatewayclient.UsageResponse{Rows: []proxygatewayclient.UsageRow{}})
	})
	srv := httptest.NewServer(mux)
	defer srv.Close()

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL})
	c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{Granularity: proxygatewayclient.GranularityTotal})

	if gotReq.URL.Query().Get("granularity") != "total" {
		t.Errorf("granularity param not sent correctly, got %q", gotReq.URL.Query().Get("granularity"))
	}
}

func TestQueryUsage_SendsPaginationParams(t *testing.T) {
	var gotReq *http.Request
	mux := http.NewServeMux()
	mux.HandleFunc("GET /api/usage", func(w http.ResponseWriter, r *http.Request) {
		gotReq = r
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(proxygatewayclient.UsageResponse{Rows: []proxygatewayclient.UsageRow{}})
	})
	srv := httptest.NewServer(mux)
	defer srv.Close()

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL})
	c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{Page: 3, PageSize: 50})

	q := gotReq.URL.Query()
	if q.Get("page") != "3" {
		t.Errorf("page param: want 3, got %q", q.Get("page"))
	}
	if q.Get("page_size") != "50" {
		t.Errorf("page_size param: want 50, got %q", q.Get("page_size"))
	}
}

func TestQueryUsage_OmitsZeroTimeParams(t *testing.T) {
	var gotReq *http.Request
	mux := http.NewServeMux()
	mux.HandleFunc("GET /api/usage", func(w http.ResponseWriter, r *http.Request) {
		gotReq = r
		w.Header().Set("Content-Type", "application/json")
		json.NewEncoder(w).Encode(proxygatewayclient.UsageResponse{Rows: []proxygatewayclient.UsageRow{}})
	})
	srv := httptest.NewServer(mux)
	defer srv.Close()

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL})
	c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{}) // all zero

	q := gotReq.URL.Query()
	for _, param := range []string{"from", "to", "proxyset", "meta", "granularity", "page", "page_size"} {
		if q.Has(param) {
			t.Errorf("param %q should be omitted when zero, but was sent as %q", param, q.Get(param))
		}
	}
}

func TestQueryUsage_ReturnsAPIError(t *testing.T) {
	srv, _ := newUsageTestServer(t, usageServerOpts{status: http.StatusBadRequest})

	c := proxygatewayclient.New(proxygatewayclient.ClientOptions{BaseURL: srv.URL})
	_, err := c.QueryUsage(context.Background(), proxygatewayclient.UsageFilter{})
	if err == nil {
		t.Fatal("expected error for 400 response")
	}
	apiErr, ok := err.(*proxygatewayclient.APIError)
	if !ok {
		t.Fatalf("expected *APIError, got %T: %v", err, err)
	}
	if apiErr.StatusCode != http.StatusBadRequest {
		t.Errorf("want StatusCode=400, got %d", apiErr.StatusCode)
	}
}

func TestQueryUsage_AllGranularityConstants(t *testing.T) {
	cases := []struct {
		g    proxygatewayclient.Granularity
		want string
	}{
		{proxygatewayclient.GranularityHour, "hour"},
		{proxygatewayclient.GranularityDay, "day"},
		{proxygatewayclient.GranularityProxyset, "proxyset"},
		{proxygatewayclient.GranularityTotal, "total"},
	}
	for _, tc := range cases {
		if string(tc.g) != tc.want {
			t.Errorf("Granularity constant: want %q, got %q", tc.want, string(tc.g))
		}
	}
}

File: proxy-gateway-client-ts/src/client.test.ts
import { describe, expect, it } from "bun:test";
import { ProxyGatewayClient } from "./client";
import type { UsageResponse } from "./types";

// ---------------------------------------------------------------------------
// Minimal test server using Bun.serve
// ---------------------------------------------------------------------------

function makeUsageServer(handler: (req: Request) => Response): { url: string; close: () => void } {
    const srv = Bun.serve({ port: 0, fetch: handler });
    return {
        url: `http://localhost:${srv.port}`,
        close: () => srv.stop(true),
    };
}

const emptyUsageResponse: UsageResponse = {
    rows: [],
    total_count: 0,
    page: 1,
    page_size: 100,
    total_pages: 1,
};

const singleRowResponse: UsageResponse = {
    rows: [
        {
            hour_ts: "2026-01-15T10:00:00Z",
            proxyset: "residential",
            affinity_params: '{"user":"alice"}',
            upload_bytes: 1024,
            download_bytes: 2048,
            total_bytes: 3072,
        },
    ],
    total_count: 1,
    page: 1,
    page_size: 100,
    total_pages: 1,
};

// ---------------------------------------------------------------------------
// queryUsage — response parsing
// ---------------------------------------------------------------------------

describe("queryUsage response parsing", () => {
    it("returns empty response when no rows", async () => {
        const srv = makeUsageServer(() => Response.json(emptyUsageResponse));
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        const result = await client.queryUsage();
        expect(result.rows).toHaveLength(0);
        expect(result.total_count).toBe(0);
        expect(result.total_pages).toBe(1);
        srv.close();
    });

    it("parses row fields correctly", async () => {
        const srv = makeUsageServer(() => Response.json(singleRowResponse));
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        const result = await client.queryUsage();
        expect(result.rows).toHaveLength(1);
        const row = result.rows[0];
        expect(row.proxyset).toBe("residential");
        expect(row.upload_bytes).toBe(1024);
        expect(row.download_bytes).toBe(2048);
        expect(row.total_bytes).toBe(3072);
        expect(row.hour_ts).toBe("2026-01-15T10:00:00Z");
        srv.close();
    });

    it("parses pagination fields", async () => {
        const resp: UsageResponse = {
            rows: [],
            total_count: 250,
            page: 2,
            page_size: 50,
            total_pages: 5,
        };
        const srv = makeUsageServer(() => Response.json(resp));
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        const result = await client.queryUsage({ page: 2, pageSize: 50 });
        expect(result.total_count).toBe(250);
        expect(result.page).toBe(2);
        expect(result.page_size).toBe(50);
        expect(result.total_pages).toBe(5);
        srv.close();
    });
});

// ---------------------------------------------------------------------------
// queryUsage — query parameter encoding
// ---------------------------------------------------------------------------

describe("queryUsage query parameters", () => {
    function captureParams(req: Request): URLSearchParams {
        return new URL(req.url).searchParams;
    }

    it("sends no params when filter is empty", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage();
        expect([...received!.keys()]).toHaveLength(0);
        srv.close();
    });

    it("sends from param", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ from: "2026-01-01T00:00:00Z" });
        expect(received!.get("from")).toBe("2026-01-01T00:00:00Z");
        srv.close();
    });

    it("sends to param", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ to: "2026-01-31T23:59:59Z" });
        expect(received!.get("to")).toBe("2026-01-31T23:59:59Z");
        srv.close();
    });

    it("sends proxyset param", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ proxyset: "datacenter" });
        expect(received!.get("proxyset")).toBe("datacenter");
        srv.close();
    });

    it("sends meta param", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ meta: '{"user":"alice"}' });
        expect(received!.get("meta")).toBe('{"user":"alice"}');
        srv.close();
    });

    it("sends granularity=hour", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ granularity: "hour" });
        expect(received!.get("granularity")).toBe("hour");
        srv.close();
    });

    it("sends granularity=day", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ granularity: "day" });
        expect(received!.get("granularity")).toBe("day");
        srv.close();
    });

    it("sends granularity=proxyset", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ granularity: "proxyset" });
        expect(received!.get("granularity")).toBe("proxyset");
        srv.close();
    });

    it("sends granularity=total", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ granularity: "total" });
        expect(received!.get("granularity")).toBe("total");
        srv.close();
    });

    it("sends page and page_size params", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage({ page: 3, pageSize: 25 });
        expect(received!.get("page")).toBe("3");
        expect(received!.get("page_size")).toBe("25");
        srv.close();
    });

    it("sends all filters together", async () => {
        let received: URLSearchParams | undefined;
        const srv = makeUsageServer((req) => {
            received = captureParams(req);
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url, apiKey: "secret" });
        await client.queryUsage({
            from: "2026-01-01T00:00:00Z",
            to: "2026-01-31T23:59:59Z",
            proxyset: "residential",
            meta: '{"platform":"ios"}',
            granularity: "day",
            page: 2,
            pageSize: 50,
        });
        expect(received!.get("from")).toBe("2026-01-01T00:00:00Z");
        expect(received!.get("to")).toBe("2026-01-31T23:59:59Z");
        expect(received!.get("proxyset")).toBe("residential");
        expect(received!.get("meta")).toBe('{"platform":"ios"}');
        expect(received!.get("granularity")).toBe("day");
        expect(received!.get("page")).toBe("2");
        expect(received!.get("page_size")).toBe("50");
        srv.close();
    });
});

// ---------------------------------------------------------------------------
// queryUsage — error handling
// ---------------------------------------------------------------------------

describe("queryUsage error handling", () => {
    it("throws on 400 response", async () => {
        const srv = makeUsageServer(
            () => new Response(JSON.stringify({ error: "invalid granularity" }), { status: 400 }),
        );
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await expect(client.queryUsage({ granularity: "weekly" as never })).rejects.toThrow("400");
        srv.close();
    });

    it("throws on 401 response", async () => {
        const srv = makeUsageServer(() => new Response(JSON.stringify({ error: "unauthorized" }), { status: 401 }));
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await expect(client.queryUsage()).rejects.toThrow("401");
        srv.close();
    });

    it("sends Authorization header when apiKey is set", async () => {
        let authHeader: string | null = null;
        const srv = makeUsageServer((req) => {
            authHeader = req.headers.get("Authorization");
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url, apiKey: "my-secret-key" });
        await client.queryUsage();
        expect(authHeader).toBe("Bearer my-secret-key");
        srv.close();
    });

    it("sends Authorization header from getToken", async () => {
        let authHeader: string | null = null;
        const srv = makeUsageServer((req) => {
            authHeader = req.headers.get("Authorization");
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({
            baseUrl: srv.url,
            getToken: () => "dynamic-token",
        });
        await client.queryUsage();
        expect(authHeader).toBe("Bearer dynamic-token");
        srv.close();
    });
});

// ---------------------------------------------------------------------------
// queryUsage — hits the correct path
// ---------------------------------------------------------------------------

describe("queryUsage URL", () => {
    it("calls /api/usage", async () => {
        let receivedPath: string | undefined;
        const srv = makeUsageServer((req) => {
            receivedPath = new URL(req.url).pathname;
            return Response.json(emptyUsageResponse);
        });
        const client = new ProxyGatewayClient({ baseUrl: srv.url });
        await client.queryUsage();
        expect(receivedPath).toBe("/api/usage");
        srv.close();
    });
});
